src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -230,10 +242,11 @@ private Pod getPodTemplate(KubernetesSlave slave, Label label) {
         url = url.endsWith("/") ? url : url + "/";
         env.add(new EnvVar("JENKINS_JNLP_URL", url + slave.getComputer().getUrl() + "slave-agent.jnlp", null));
 
-        for (PodEnvVar podEnvVar :template.getEnvVars()) {
-            env.add(new EnvVar(podEnvVar.getKey(), podEnvVar.getValue(), null));
+        if(template.getEnvVars()!=null) {
+	        for (PodEnvVar podEnvVar :template.getEnvVars()) {
+	            env.add(new EnvVar(podEnvVar.getKey(), podEnvVar.getValue(), null));
+	        }
```

Comment:
better if this goes into its separate commit as it is unrelated to the main commit